tests/core/test_signal.py

```diff
@@ -281,3 +282,39 @@ async def test_subscription_logs(caplog):
     assert "Making subscription" in caplog.text
     cache.close()
     assert "Closing subscription on source" in caplog.text
+
+
+async def test_signal_lazily_connects(RE):
+    mock_signal_rw = soft_signal_rw(
+        int, "pva://this is an initial value not a pva pv", name="mock_signal"
+    )
+    await mock_signal_rw.connect(mock=False)
+    RE(ensure_connected(mock_signal_rw, mock=True))
+    assert (
+        mock_signal_rw._connect_task
+        and mock_signal_rw._connect_task.done()
+        and not mock_signal_rw._connect_task.exception()
+    )
+
+
+async def test_signal_mock_and_back_again(RE):
+    mock_signal_rw = soft_signal_rw(int, "pva://mock_signal", name="mock_signal")
+    assert not mock_signal_rw._connect_task
+    await mock_signal_rw.connect(mock=False)
+    assert isinstance(mock_signal_rw._backend, SoftSignalBackend)
+    assert mock_signal_rw._connect_task
+    # todo is explicit disconnection needed?
+    await mock_signal_rw.connect(mock=True)
+    assert isinstance(mock_signal_rw._backend, MockSignalBackend)
+
+
+async def test_connect_fails_then_next_connect_succeeds():
+    raise AssertionError("Not implemented")
+    mock_signal_rw = soft_signal_rw(int, "pva://mock_signal", name="mock_signal")
```

Comment:
```suggestion
    mock_signal_rw = soft_signal_rw(int, 0, name="mock_signal")
```

In a `soft_signal_rw` that is the value, it doesn't take a PV.

If you want this to fail then perhaps you could make a `pva_signal_backend` so the initial connect fails (since the PV won't exist), then afterwards swap out the backend to a mocked one.

Reply:
applied